CS483/CS483/Kartaclysm/Services/IO/InputActionMapping.cpp

```diff
@@ -109,59 +109,70 @@ namespace Kartaclysm
 			return;
 		}
 
-		HeatStroke::Event* pEvent = new HeatStroke::Event("PlayerInput_" + p_iPlayer);
+		// Helpful variables to avoid repetition
+		std::string strPlayerIdentifier = "P" + std::to_string(p_iPlayer) + "_";
+		HeatStroke::EventManager* pEventManager = HeatStroke::EventManager::Instance();
 
-		if (p_iGLFWJoystick == GLFW_JOYSTICK_LAST + 1)
+		if (p_iGLFWJoystick == GLFW_JOYSTICK_LAST + 1) // Keyboard
 		{
-			// Keyboard keys
+			// Helpful variables to avoid repetition
 			HeatStroke::KeyboardInputBuffer* pBuffer = HeatStroke::KeyboardInputBuffer::Instance();
-			pEvent->SetIntParameter("Accelerate", static_cast<int>(pBuffer->IsKeyDown((*m_pInputMap)[Input::eKeyboard][Racer::eAccelerate])));
-			pEvent->SetIntParameter("Brake", static_cast<int>(pBuffer->IsKeyDown((*m_pInputMap)[Input::eKeyboard][Racer::eBrake])));
-			pEvent->SetIntParameter("Slide", static_cast<int>(pBuffer->IsKeyDown((*m_pInputMap)[Input::eKeyboard][Racer::eSlide])));
-			pEvent->SetIntParameter("Driver1", static_cast<int>(pBuffer->IsKeyDown((*m_pInputMap)[Input::eKeyboard][Racer::eDriverAbility1])));
-			pEvent->SetIntParameter("Driver2", static_cast<int>(pBuffer->IsKeyDown((*m_pInputMap)[Input::eKeyboard][Racer::eDriverAbility2])));
-			pEvent->SetIntParameter("Kart1", static_cast<int>(pBuffer->IsKeyDown((*m_pInputMap)[Input::eKeyboard][Racer::eKartAbility1])));
-			pEvent->SetIntParameter("Kart2", static_cast<int>(pBuffer->IsKeyDown((*m_pInputMap)[Input::eKeyboard][Racer::eKartAbility2])));
-			pEvent->SetIntParameter("Pause", static_cast<int>(pBuffer->IsKeyDown((*m_pInputMap)[Input::eKeyboard][Racer::ePause])));
-
-			// For compatibility with joysticks, turn is a float value: Right adds +1.0f, Left adds -1.0f
-			float fTurn = static_cast<float>(pBuffer->IsKeyDown((*m_pInputMap)[Input::eKeyboard][Racer::eRight]));
-			fTurn -= static_cast<float>(pBuffer->IsKeyDown((*m_pInputMap)[Input::eKeyboard][Racer::eLeft]));
-			pEvent->SetFloatParameter("Turn", fTurn);
+			ActionMap mActionMap = (*m_pInputMap)[Input::eKeyboard];
+
+			// Iterate ability and pause actions
+			if (pBuffer->IsKeyDown(mActionMap[Racer::eDriverAbility1]))
+			{
+				pEventManager->TriggerEvent(new HeatStroke::Event(strPlayerIdentifier + "Driver1"));
```

Comment:
I would change the end of the string from just "DriverX" to "DriverAbilityX", to make it more clear what the event is.